crates/storage/provider/src/providers/static_file/manager.rs

```diff
@@ -448,6 +452,161 @@ impl StaticFileProvider {
         Ok(())
     }
 
+    /// Ensures that any broken invariants return a pipeline target to unwind to.
+    ///
+    /// Two types of consistency checks are done for:
+    ///
+    /// 1) When a static file fails to commit but the underlying data was changed.
+    /// 2) When a static file was committed, but the required database transaction was not.
+    ///
+    /// For 1) it can self-heal. Opening a writer to this segment will automatically do that.
+    /// For 2) the invariants below are checked, and if broken, require a pipeline unwind to heal.
+    ///
+    /// For each static file segment:
+    /// * the corresponding database table should overlap or have continuity in their keys
+    ///   ([TxNumber] or [BlockNumber]).
+    /// * its highest block should match the stage checkpoint block number if it's equal or higher
+    ///   than the corresponding database table last entry.
+    ///
+    /// Returns a [`Option`] of [`PipelineTarget::Unwind`] if any healing is required.
+    pub fn check_consistency<TX: DbTx>(
+        &self,
+        provider: &DatabaseProvider<TX>,
+        has_receipt_pruning: bool,
+    ) -> ProviderResult<Option<PipelineTarget>> {
+        let mut unwind_target: Option<BlockNumber> = None;
+        let mut update_unwind_target = |new_target: Option<BlockNumber>| {
+            let new_target = new_target.unwrap_or_default();
```

Comment:
This `unwrap_or_default` seems correct, because we need to unwind to `0` if we pass `None` as a new target. I would prefer passing just the `BlockNumber` here instead, and do the `unwrap_or_default()` only in the call where `None` is possible.